tools/settings.py

```diff
@@ -1,54 +1,34 @@
 """
 mbed SDK
-Copyright (c) 2011-2013 ARM Limited
+Copyright (c) 2016 ARM Limited
+
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
-    http://www.apache.org/licenses/LICENSE-2.0
+
+http://www.apache.org/licenses/LICENSE-2.0
+
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 """
 
-from os.path import join, abspath, dirname
+from os import getenv
+from os.path import join, abspath, dirname, exists
 import logging
 
 ROOT = abspath(join(dirname(__file__), ".."))
 
-# These default settings have two purposes:
-#    1) Give a template for writing local "mbed_settings.py"
-#    2) Give default initialization fields for the "toolchains.py" constructors
 
 ##############################################################################
-# Build System Settings
+# Toolchains and Build System Settings
 ##############################################################################
 BUILD_DIR = abspath(join(ROOT, ".build"))
 
-# ARM
-armcc = "standalone" # "keil", or "standalone", or "ds-5"
-
-if armcc == "keil":
-    ARM_PATH = "C:/Keil_v5/ARM/ARMCC"
-    ARM_BIN = join(ARM_PATH, "bin")
-    ARM_INC = join(ARM_PATH, "incldue")
-    ARM_LIB = join(ARM_PATH, "lib")
-
-elif armcc == "standalone":
-    ARM_PATH = "C:/Program Files (x86)/ARM_Compiler_5.06u1"
-    ARM_BIN = join(ARM_PATH, "bin")
-    ARM_INC = join(ARM_PATH, "include")
-    ARM_LIB = join(ARM_PATH, "lib")
-
-elif armcc == "ds-5":
-    ARM_PATH = "C:/Program Files (x86)/DS-5"
-    ARM_BIN = join(ARM_PATH, "bin")
-    ARM_INC = join(ARM_PATH, "include")
-    ARM_LIB = join(ARM_PATH, "lib")
-
-ARM_CPPLIB = join(ARM_LIB, "cpplib")
-MY_ARM_CLIB = join(ARM_PATH, "lib", "microlib")
+# ARM Compiler 5
+ARM_PATH = "C:/Program Files (x86)/ARM_Compiler_5.06u1"
```

Comment:
one day i'll convince you to set this path for Keil MDK rather than just the compiler... one day...